maze_transformer/generation/lattice_maze.py

```diff
@@ -306,17 +364,426 @@ def from_tokens(cls, tokens: list[str]) -> "LatticeMaze":
 
         return cls.from_adj_list(adj_list)
 
+    # ============================================================
+    # to and from pixels
+    # ============================================================
+    def _as_pixels_bw(self) -> BinaryPixelGrid:
+        assert self.lattice_dim == 2, "only 2D mazes are supported"
+        # Create an empty pixel grid with walls
+        pixel_grid: Int[np.ndarray, "x y"] = np.full(
+            (self.grid_shape[0] * 2 + 1, self.grid_shape[1] * 2 + 1),
+            False,
+            dtype=np.bool_,
+        )
+
+        # Set white nodes
+        pixel_grid[1::2, 1::2] = True
+
+        # Set white connections (downward)
+        for i, row in enumerate(self.connection_list[0]):
+            for j, connected in enumerate(row):
+                if connected:
+                    pixel_grid[i * 2 + 2, j * 2 + 1] = True
+
+        # Set white connections (rightward)
+        for i, row in enumerate(self.connection_list[1]):
+            for j, connected in enumerate(row):
+                if connected:
+                    pixel_grid[i * 2 + 1, j * 2 + 2] = True
+
+        return pixel_grid
+
+    def as_pixels(
+        self,
+        show_endpoints: bool = True,
+        show_solution: bool = True,
+    ) -> PixelGrid:
+        if show_solution and not show_endpoints:
+            raise ValueError("show_solution=True requires show_endpoints=True")
+        # convert original bool pixel grid to RGB
+        pixel_grid_bw: BinaryPixelGrid = self._as_pixels_bw()
+        pixel_grid: PixelGrid = np.full(
+            (*pixel_grid_bw.shape, 3), PixelColors.WALL, dtype=np.uint8
+        )
+        pixel_grid[pixel_grid_bw == True] = PixelColors.OPEN
+
+        if self.__class__ == LatticeMaze:
+            return pixel_grid
+
+        # set endpoints for TargetedLatticeMaze
+        if self.__class__ == TargetedLatticeMaze:
```

Comment:
This is a bit sus. Would it be better to have an `as_pixels` on `LatticeMaze` that just returns the basic version, and `as_pixels` on TargettedMaze and SolvedMaze that override it and provide the extra functionality?

Reply:
There are a lot of type errors in the rest of this method because you're trying to do stuff that isn't possible on `LatticeMaze`

Reply:
(not sure if my solution makes sense - I think it is good to locate the class-specific functionality with the appropriate class, but not 100% sure what the most pythonic version of that looks like)

Reply:
discussed this during engineering meeting, but for context: I did originally have 3 separate functions, but that was very messy and resulted in a lot of repeated code. Type checking might complain, an `assert` of the type after the `if self.__class__ == WhicheverClass` might help